internal/cmd/grants.go

```diff
@@ -296,7 +302,13 @@ func getIDByName(client *api.Client, name string, identityType identityType) (ui
 
 		switch len(identities) {
 		case 0:
-			return "", fmt.Errorf("No identity of name %s exists", name)
+			created, err := CreateIdentity(name)
+			if err != nil {
+				return "", err
+			}
+			fmt.Printf("New unlinked identity %q added to Infra\n", name)
+
+			id = uid.NewIdentityPolymorphicID(created.ID)
```

Comment:
removing grants for identities can inadvertently add that identity. 
also cli should probably confirm before adding this?